array-destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +71,20 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+
+let gryffindorStudentNames = hogwarts.filter(({house}) => {
+  return house === "Gryffindor";
+});
+for (const student of gryffindorStudentNames) {
+  let { firstName, lastName } = student;
+  console.log(firstName, lastName);
+}
+
+let teachersWithPets = hogwarts.filter(({occupation, pet}) => {
+  return (occupation === "Teacher") && (pet !== null);
+});
+for (const teacher of teachersWithPets) {
+  let { firstName, lastName } = teacher;
+  console.log(firstName, lastName);
+}
```

Comment:
Well done finding a valid solution for both problems, a few notes to improve these solutions.

You are using the filter method and assigning the results to a new variable. While this is an acceptable approach, you have used the let declaration when this variable is not reassigned at any stage, so const would have been the correct declaration to use.

Also, you are using two loops for each solution, when it can be simplified to use 1. This can be achieved using the Array.forEach method or just a for loop which you already have. Then using a condition to check the object values before logging the character name.

Reply:
Hey Jack!
Thank you so much for your valuable feedback! 
I will make the changes now.

Reply:
I made the changes according to you second suggestion.